.travis.yml

```diff
@@ -29,52 +21,122 @@ compiler:
           compiler: clang-3.6
-          env: CONFIG_OPTS="no-asm enable-ubsan enable-rc5 enable-md2 enable-ssl3 enable-ssl3-method enable-nextprotoneg -fno-sanitize=alignment no-shared -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION"
+          env: CONFIG_OPTS="no-asm enable-ubsan enable-rc5 enable-md2 enable-ssl3 enable-ssl3-method enable-nextprotoneg -fno-sanitize=alignment no-shared -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION" EXTENDED_TEST="yes"
         - os: linux
+          addons:
+              apt:
+                  packages:
+                      - clang-3.6
+                  sources:
+                      - llvm-toolchain-precise-3.6
+                      - ubuntu-toolchain-r-test
           compiler: clang-3.6
-          env: CONFIG_OPTS="no-asm enable-asan enable-rc5 enable-md2 no-shared -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION"
+          env: CONFIG_OPTS="no-asm enable-asan enable-rc5 enable-md2 no-shared -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION" EXTENDED_TEST="yes"
         - os: linux
+          addons:
+              apt:
+                  packages:
+                      - gcc-5
+                      - g++-5
+                  sources:
+                      - ubuntu-toolchain-r-test
           compiler: gcc-5
-          env: CONFIG_OPTS="no-asm enable-ubsan enable-rc5 enable-md2 -DPEDANTIC"
+          env: CONFIG_OPTS="no-asm enable-ubsan enable-rc5 enable-md2 -DPEDANTIC" EXTENDED_TEST="yes"
         - os: linux
+          addons:
+              apt:
+                  packages:
+                      - gcc-5
+                      - g++-5
+                  sources:
+                      - ubuntu-toolchain-r-test
           compiler: gcc-5
           env: CONFIG_OPTS="--strict-warnings enable-tls1_3" COMMENT="Move to the BORINGTEST build when interoperable"
         - os: linux
+          addons:
+              apt:
+                  packages:
+                      - binutils-mingw-w64
+                      - gcc-mingw-w64
           compiler: i686-w64-mingw32-gcc
-          env: CONFIG_OPTS="no-stdio" BUILDONLY="yes"
+          env: CONFIG_OPTS="no-stdio" BUILDONLY="yes" EXTENDED_TEST="yes"
         - os: linux
+          addons:
+              apt:
+                  packages:
+                      - binutils-mingw-w64
+                      - gcc-mingw-w64
+                      - wine
           compiler: i686-w64-mingw32-gcc
-          env: CONFIG_OPTS="no-pic"
+          env: CONFIG_OPTS="no-pic" EXTENDED_TEST="yes"
         - os: linux
+          addons:
+              apt:
+                  packages:
+                      - binutils-mingw-w64
+                      - gcc-mingw-w64
+                      - wine
           compiler: x86_64-w64-mingw32-gcc
-          env: CONFIG_OPTS="no-pic"
+          env: CONFIG_OPTS="no-pic" EXTENDED_TEST="yes"
     exclude:
         - os: linux
           compiler: clang
         - os: osx
           compiler: gcc
 
 before_script:
+    - env
```

Comment:
Just in case for reference. If choose to take this path, env can be removed *after* it repo-commit. As you can see `[extended tests]` are evaluated when TRAVIS_EVENT_TYPE is pull_request. This asks for question what is TRAVIS_EVENT_TYPE when repo-commit is performed. Once we know env can be removed...